### PR TITLE
simple model scorers to accept averaged scorers 

### LINKED ISSUE #2235: [BUG] SimpleModelComparison binary classification scorer error
**Describe the bug**
When I am doing binary classifier model with simple model comparsion checks:
1. The metric I used(https://docs.deepchecks.com/stable/user-guide/general/metrics_guide.html#alternative-metrics): 
image
2. No matter what metric I used(accuracy, f1, roc_auc...), the error always exists:
image
3. When I use debug mode, the task_type is binary, and scorer(model_instance, test_dataset) returns a numpy float instead of a dict, so it throws an exception:
image

**Expected behavior**
`scorer(model_instance, test_dataset) ` should returns a dict-like object

**Screenshots**

**Environment (please complete the following information):**
 - OS: Mac OS
 - Python Version: 3.8
 - Deepchecks Version: 0.10

**Additional context**

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #2235

### CODE REVIEW

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 1]
```diff
@@ -181,20 +182,32 @@ def run_logic(self, context: Context) -> CheckResult:
...
+                        model_dict[None][model_type] = scorer_value
                         if context.with_display:
                             display_array.append([model_name,
                                                   model_type,
-                                                  class_score,
+                                                  scorer_value,
                                                   scorer.name,
-                                                  class_value,
-                                                  class_counts[class_value]
+                                                  None,
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:193]: change to 'All'

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 2]
```diff
@@ -181,20 +182,32 @@ def run_logic(self, context: Context) -> CheckResult:
...
+                                                  None,
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:194]: can place here number of samples in the dataset

File: tests/tabular/checks/model_evaluation/simple_model_comparison_test.py [hunk 3]
```diff
@@ -68,6 +68,18 @@ def test_classification_binary_string_labels(iris_binary_string_split_dataset_an
...
 
 
+def test_classification_binary_string_labels_custom_scorer(iris_binary_string_split_dataset_and_model):
+    # Arrange
+    train_ds, test_ds, clf = iris_binary_string_split_dataset_and_model
+    check = SimpleModelComparison(scorers=[get_scorer('f1')])
+    # Act X
+    result = check.run(train_ds, test_ds, clf).value
+    # Assert
+    assert_that(result, equal_to({'scorers_perfect': {'f1_score': 1.0},
```
Comment (Nadav-Barak) [inline tests/tabular/checks/model_evaluation/simple_model_comparison_test.py:78]: If can be done easily, remove from the result.value the redundant dict, i.e make it 
'f1_score': {'Origin': 0.9411764705882353, 'Simple': 0.0}}

Comment (JKL98ISR) [reply tests/tabular/checks/model_evaluation/simple_model_comparison_test.py:78]: might destroy stuff

**Review by Nadav-Barak — APPROVED** [review]

**Pull request merged**